Assets/Scripts/Managers/DayCycleManager.cs

```diff
@@ -48,6 +48,12 @@ void Start()
 		}
 	}
 
+	void OnValidate()
+	{
+		CancelInvoke( "StartMidnightOverlay" );
+		Invoke( "StartMidnightOverlay", ( _dayCycleLength - _blackoutDuration ) - _currentTime );
```

Comment:
This calculation is done in multiple places, so that indicates it should probably be its own function.